Sources/ShapedArray/Operations/pack.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jaap Wijnen on 10/08/2022.
+//
+
+extension ShapedArray {
+    static func pack(_ values: [ShapedArray], axis: Int = 0) -> ShapedArray {
+
+        precondition(!values.isEmpty, "Cannot pack empty array of ShapedArrays.")
+        let shape = values.first!.shape
+        precondition(axis >= 0 && axis <= shape.count, "axis = \(axis) is not within [0, \(shape.count)]")
+        assert(!values.map { $0.shape }.contains { $0 != shape }, "Shapes of all inputs must match: \(shape).")
```

Comment:
`assert` is rarely used in practice, as it doesn't throw an error in release mode. However, this is check is crucial for runtime performance. How about we document why we're using an assertion instead of a precondition?

Also, the map operation can be optimized by doing:

```swift
values.contains(where: { $0.shape != shape })
```

I prefer to use `where` because it's semantically appropriate. We should to a code scan of Apple's swift repos and S4TF, and see how often the argument label is shown.